Framework/DataHandling/test/LoadEmptyInstrumentTest.h

```diff
@@ -136,7 +149,92 @@ class LoadEmptyInstrumentTest : public CxxTest::TestSuite {
     check_workspace_detectors(output, 64);
   }
 
+  void testLoadFromNexus() {
+    g_log.notice("\ntestLoadFromNexus...");
+
+    LoadEmptyInstrument loader;
+
+    TS_ASSERT_THROWS_NOTHING(loader.initialize());
+    TS_ASSERT(loader.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("Filename", "PG3_46577.nxs.h5"));
+    TS_ASSERT_THROWS_NOTHING(inputFile = loader.getPropertyValue("Filename"));
+    wsName = "LEIT_LoadFromNexus";
+    TS_ASSERT_THROWS_NOTHING(loader.setPropertyValue("OutputWorkspace", wsName));
+
+    std::string result;
+    TS_ASSERT_THROWS_NOTHING(result = loader.getPropertyValue("Filename"));
+    TS_ASSERT_EQUALS(result, inputFile);
```

Comment:
The test of "Filename" is performed twice

Reply:
The key thing here is that the Filename string is not exactly what was set originally.  The original string is simply a file name. Then Mantid uses its intelligence and finds the full path of that file, and that's what ends up in the Filename property.

Reply:
It's a full path. For example, if we set `"PG3_46577.nxs.h5"`, then `getPropertyValue` would retrieve a string ending with `"../build/ExternalData/Testing/Data/UnitTest/PG3_46577.nxs.h5"`.  
I agree with your other comment that unless there is reason to suspect that `setPropertyValue("OutputWorkspace")` changes `Filename`, there is no reason to test that `result = loader.getPropertyValue("Filename")` is equal to `inputFile`.

Reply:
I ended up removing all those round-trip checks: `setPropertyValue`->`getPropertyValue`. I believe they are unnecessary in this context.